### ISSUE TITLE
roachtest: change mixedversion tests to perform "skip version" upgrades

### ISSUE THREAD

**Issue by RaduBerinde** [cockroachdb/cockroach#121957]
We should make mixedversion tests (like `version-upgrade`) test version skipping, e.g. upgrade from 23.1 directly to 24.1.

PR #106114 was an attempt at this, but it did not get completed.

CC @celiala @rail 

Jira issue: CRDB-37643

**Comment by renatolabs** [comment]
Could someone remind me what are the possible "skip version" scenarios? 

* 23.1 -> 24.1
* 23.2 -> 24.2

Alternatively, would it be valid to say (for now) that every version starting from X (maybe 23.1) can perform a "skip version" upgrade?

**Comment by RaduBerinde** [comment]
Correct, 23.1 -> 24.1 and 23.2 -> 24.1 (which is equivalent to saying we can skip starting with 23.1).

This proposed test would try to upgrade from releases with version `clusterversion.MinSupported` to `Latest`.  Equivalently, it could check if `MinSupported < PreviousRelease`, in which case, it would select a predecessor of a predecessor.

**Comment by srosenberg** [comment]
I also did a couple of manual runs of `schemachange/mixed-versions` by forcing to skip the penultimate version [1]. It seems the current (upgrade) machinery supports "version skipping", but we still need to add it as an optional feature to the mixed-version framework.

[1] https://github.com/srosenberg/cockroach/commit/ab2b91dc3c0108e79df4bfee46ba91f105dccdc8

**Comment by renatolabs** [comment]
I have a draft PR to implement this, but it's not quite ready to be used: master's version is still at 24.1.X, which means a "skip-version" upgrade from master would start from a release in the 23.1 series. However, `master`'s MinSupported was already bumped to 23.2, so that wouldn't work.

We could bump master to 24.2, but then its immediate predecessor would be 24.1, and we don't have any rcs for that release yet.

TLDR: I'll put that PR on pause until we have an rc for 24.1 (_and_ are able to generate fixtures for 24.1, _and_ are able to bump master to 24.2).

**Comment by renatolabs** [comment]
Why was this closed? #125869 hasn't merged yet. Reopening.
